cfme/configure/settings.py

```diff
@@ -23,109 +25,173 @@ class TimeProfileAddForm(View):
         for row in rows:
             if row.description.text == self.description:
                 row[0].check()
-        view.timeprofile_form.configuration.item_select('Edit selected Time Profile')
-        changed = view.timeprofile_form.fill({
+        view.time_profile_form.configuration.item_select('Edit selected Time Profile')
+        changed = view.time_profile_form.fill({
             'description': updates.get('description'),
             'scope': updates.get('scope'),
             'days': updates.get('days'),
             'hours': updates.get('hours'),
             'timezone': updates.get('timezone'),
         })
         if changed:
-            view.timeprofile_form.save_edit_button.click()
+            view.save_edit_button.click()
             view.flash.assert_message(
                 'Time Profile "{}" was saved'.format(updates.get('description', self.description)))
 
     def copy(self, name=None):
-        view = navigate_to(self, 'All')
-        rows = view.timeprofile_form.table
+        """
+        This method performs the copy of the provided time profile object.
+        Args:
+            name (str) : It's the name of the new copied time_profile.
+
+        return: It returns the object of the copied time_profile.
+        """
+
+        view = navigate_to(self.parent, 'All')
+        rows = view.time_profile_form.table
         for row in rows:
             if row.description.text == self.description:
                 row[0].check()
-        view.timeprofile_form.configuration.item_select('Copy selected Time Profile')
+        view.time_profile_form.configuration.item_select('Copy selected Time Profile')
         if name is not None:
-            new_timeprofile = Timeprofile(description=name, scope=self.scope)
-            changed = view.timeprofile_form.fill({
+            new_time_profile = self.parent.instantiate(description=name, scope=self.scope)
+            changed = view.time_profile_form.fill({
                 'description': name,
                 'scope': self.scope,
             })
         else:
-            new_timeprofile = Timeprofile(description="{} copy".format(self.description),
-                                          scope=self.scope)
-            changed = view.timeprofile_form.fill({
+            new_time_profile = self.parent.instantiate(
+                description="{} copy".format(self.description),
+                scope=self.scope
+            )
+            changed = view.time_profile_form.fill({
                 'description': "{} copy".format(self.description),
                 'scope': self.scope,
             })
 
         if changed:
-            view.timeprofile_form.save_button.click()
-        return new_timeprofile
+            view.time_profile_form.save_button.click()
+        return new_time_profile
+
+
+@attr.s
+class TimeProfileCollection(BaseCollection):
+
+    ENTITY = TimeProfile
+
+    def create(self, description, scope, days, hours, timezone, cancel=False):
```

Comment:
Why are the args required in `collection.create()` but optional in the entity's attr.ib() definitions?  They should match.  See other implementations of entity/collection, such as `User`/`UserCollection` in `cfme.configure.access_control`